### DISCUSSION TITLE
StreamFX 0.12.0 Alpha 151

### DISCUSSION THREAD

**Discussion by Xaymar** [Xaymar/obs-stream-effects#972] [Announcements]
# ⚠ Preview Release not for Production! ⚠
This release is meant for testing in isolated conditions and early adopters only, and should never be used in production. If you use one of these releases, all promises of support, warranty, functionality, and similar are immediately void and if things break you are on your own. [The latest production ready release can be found here](https://github.com/Xaymar/obs-StreamFX/releases/latest).

# Please read the [Installation Guide](https://github.com/Xaymar/obs-StreamFX/wiki/Installation) and System Requirements!
The installation of StreamFX requires that you follow certain instructions, as well as fulfill a minimum hardware and software requirement. Please ensure that you have read and verified both before asking for help, as many issues can be fixed by reading both of them in their entirety.

# Support StreamFX on Github Sponsors or Patreon!
[<img align="left" alt="GitHub Sponsors" src="https://user-images.githubusercontent.com/437395/106463119-48412400-6497-11eb-9934-baed8e0bdd73.png" height="70px"/>](https://github.com/sponsors/xaymar) [<img align="right" alt="Patreon" src="https://user-images.githubusercontent.com/437395/106462708-bd602980-6496-11eb-8f35-038577cf8fd7.jpg" height="70px"/>](https://patreon.com/xaymar) The StreamFX project is only possible through your support, either as a [translator](https://github.com/Xaymar/obs-StreamFX/issues/36) or as a Supporter on either [Patreon](https://patreon.com/xaymar) or [Github](https://github.com/sponsors/xaymar). Your support helps keep StreamFX free from Ads, even if it's just translating a few lines per month, or 1$ per month. If continuous support isn't your thing, maybe a one-time donation via [PayPal](https://paypal.me/xaymar) is?

# What's Changed
## New Release process: Alpha -> Beta -> Candidate -> Release!
The old two stage system didn't work out quite as I'd have hoped: It didn't communicate the dangers of Testing well, and did not differentiate between different stability levels. To solve this I've adopted a new release process now available in StreamFX, which follows the usual development cycle of Software itself: Alpha -> Beta -> Candidate -> Release. The further right you go in that graph, the more stable software is supposed to be, at the cost of new features and changes. The further left you go, the more up-to-date things will be, but you sacrifice stability and safety.

### Shader Texture Inputs no longer corrupt Scene Collections
For quite some time, Shader Texture Inputs were haunted by ghosts from the past, also known as work-arounds and deprecated code. This resulted in Shader Texture Inputs keeping stale references around and requiring an OBS Studio restart to actually update. Several code modernization cycles later, this is now fixed and Shader Texture Inputs immediately update as expected - while also no longer keeping stale references around, so your Scene Collections are safe again.

### Dynamic Mask no longer self-references and double-frees its Input mask
This should have gone up in flames way sooner than it did, and yet it was almost completely invisible until OBS Studio 27.1. The bug dates as far back as [v0.7](https://github.com/Xaymar/obs-StreamFX/releases/tag/0.7.0) and is pretty severe: The Dynamic Mask filter would increment their own reference count every frame if no *Input* was selected, and would double-free the Input without clearing the pointer to it. And somehow, this didn't explode until now.

### Loadable Bundles on MacOS
With OBS Studio 28.0 came the introduction of Loadable Bundles, a variation of MacOS's Bundle format. This format offers a lot of advantages for almost no drawbacks, and StreamFX from now on uses this packaging format on MacOS. As a bonus, FFmpeg Encoders should now be available on MacOS too.

#### ⚠ This change requires that you cleanly remove old versions of StreamFX first!
As the packaging and installation method has changed, please follow the [Uninstallation guide](https://github.com/Xaymar/obs-StreamFX/wiki/Uninstallation) to clean up any left over old files. Failure to do so will result in undefined behavior and may render OBS Studio entirely unusable. **You have been warned!**

### Improved System-wide installation on Windows
Investigation into how OBS Studio actually loads plugins revealed a change that had gone under the radar entirely. Since [OBS Studio v0.15.0](https://github.com/obsproject/obs-studio/releases/tag/0.15.0) it is possible to load plugins from `%ProgramData%\obs-studio\plugins`, which is a much safer option and now the only supported System-wide installation mode for StreamFX. This functionality has been so hidden that even the official [obs-plugintemplate](https://github.com/obsproject/obs-plugintemplate/blob/056422b2614fed1037bc1f19e0e468165e3068ea/cmake/bundle/windows/installer-Windows.iss.in) doesn't make use of it. An added bonus is that this should support Steam users of OBS Studio.

#### ⚠ This change requires that you cleanly remove old versions of StreamFX first!
As the packaging and installation method has changed, please follow the [Uninstallation guide](https://github.com/Xaymar/obs-StreamFX/wiki/Uninstallation) to clean up any left over old files. Failure to do so will result in undefined behavior and may render OBS Studio entirely unusable. **You have been warned!**

### User-only installation on Windows
StreamFX can now be installed for individual users, instead of installing for everyone on the system! This is possible thanks to [a PR](https://github.com/obsproject/obs-studio/pull/4067) by @abcdw, which allows us to set two environment variable to install and load plugins from `%LocalAppData%\Programs\obs-studio\plugins` in addition to all the other locations. These installations will conflict with a System-wide install and as such can't coexist, but may still work with the Steam release of OBS Studio.

### Avid DNxHR Encoder (via FFmpeg) by @IceStormNG
A new contender for the visually lossless intermediate codec arrives: Avid DNxHR! Thanks to work done by @IceStormNG, it is now possible to use this encoder without configuration trickery and directly configure it with a proper user interface. As this encoder is based on the real codec specification, it may perform better than the Apple ProRes encoder while being way more compatible with video editing software.

### Split Framerate support for FFmpeg Encoders
All FFmpeg Encoders now support encoding a different framerate than OBS Studio itself is running at. This allows for streaming at 30 FPS while recording at 60 FPS, or even recording at 120 FPS, or other strange things. Note that this is quite literally breaking several specifications, so use at your own risk - if things break, be happy that at least they broke with a noticable impact!

### Universal Binaries for x86 & ARM64 support on MacOS
Users on ARM64 devices can now enjoy native ARM64 binaries for less CPU usage, and as such more overall performance on CPU focused tasks. While StreamFX is mostly GPU workloads, some of the internal code does actually do very heavy lifting that benefits from this.

### Initial sRGB and HDR support for Sources/Filters/Transitions
With HDR support in OBS Studio finally being available, it was necessary to finally tackle the issue of sRGB and HDR support once and for all. Many Sources/Filters/Transitions will get proper sRGB and HDR support over future releases to catch up with OBS Studio. You can check the *Version Information* on the wiki to see if a certain feature supports sRGB or HDR yet.

### Deprecation of the AMD AMF (via FFmpeg) encoders
With OBS Studio 28.0 the old obs-amd-encoder project is finally laid to rest and a new AMD AMF integration is provided. This AMD AMF integration is technically and functionally superior to both the old obs-amd-encoder and FFmpegs AMD AMF integration. All users are urged to migrate as soon as possible, as no guarantees are made for how long this deprecated encoder will remain. Additionally, users on Linux should put their focus on the open Pull Requests for Linux support ([1](https://github.com/obsproject/obs-studio/pull/7206), [2](https://github.com/obsproject/obs-studio/pull/4974)).

### Deprecation of the AV1 (via AOM) Encoder
With OBS Studio 27.2.4, the SVT-AV1 and AOM-AV1 encoders were integrated into OBS Studio directly. This integration is functionally identical, and as such replaces the StreamFX provided AOM-AV1 encoder. All users are urged to migrate as soon as possible, as no guaratees are made for how long this deprecated encoder will remain.

## [Detailed Changelog](https://github.com/Xaymar/obs-StreamFX/compare/0.11.0...0.12.0a151)

<details><summary>0.12.0a151</summary>

> * Implement dynamically scaled Thread Pool by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/933
> * Save configuration after ~100ms have passed by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/934
> * templates/windows/installer: Don't create Icons in Portable mode by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/936
> * templates/windows/installer: Implement `/Portable` installer flag for CLI users by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/935
> * cmake, templates: Remove code signing integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/938
> * Don't show path selection for System & User mode by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/937
> * Fix some Use-After-Free issues with the D3D11 integration with FFmpeg by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/953
> * encoder/ffmpeg: Remove usage of `avcodec_close` by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/952
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/932
> * Implement support for Color Spaces by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/968
> * Dynamic Mask: Implement sRGB and HDR support, and fix #819 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/967
</details>

<details><summary>0.12.0a134</summary>

> * cmake: Require nlohmann-json when building the front-end by @rmnvgr in https://github.com/Xaymar/obs-StreamFX/pull/914
> * encoder/ffmpeg: Assume extra data and sei are present on first frame by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/919
> * encoder/ffmpeg: Add split framerate with integer fractions by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/921
> * encoder/ffmpeg: Improve software scaling quality by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/925
> * Add support for HDR formats by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/922
> * encoder/handler/prores_aw: Don't enable Key-Frame options by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/926
> * encoder/handler/dnxhd: Use endian-independent formats by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/927
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/915
> * Improve Windows installer with proper System install location, Current User install method, and better UI by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/930
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/928
</details>

<details><summary>0.12.0a117</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/907
> * Improve Windows and Mac bundling by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/912
</details>

<details><summary>0.12.0a106</summary>

> * #881: Texture inputs to Shaders are unstable by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/884
> * Deprecate AMD AMF encoders by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/885
> * Deprecate AOM AV1 encoder by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/886
> * Add missing deprecation notices by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/889
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/867
> * cmake: Fix bad version strings by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/887
> * project: Updating contributing guidelines by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/894
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/890
> * project: Updating contributing guidelines by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/895
> * Fix a huge number of warnings by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/893
> * third-party: Update obs-studio to 28.0.0 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/901
> * code: Fix GCC warning diagnostic pragmas by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/902
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/897
</details>

<details><summary>0.12.0a77</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/817
> * Improve GitHub integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/841
> * project: Fix incorrect tabs in YAML by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/842
> * Improve metadata generation by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/837
> * cmake: Add support for Qt6 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/866
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/832
> * Further improve CI and Github integration by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/843
> * ci: Remove Code Signing with self-signed certificate by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/873
> * Fix compilation errors with Debug and/or Profiling by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/876
> * Deduplicate CI build pipeline by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/874
> * Remove Auto-Dependency System by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/870
> * Working on source code [Refactor] by @GermanAizek in https://github.com/Xaymar/obs-StreamFX/pull/836
> * obs/gs/effect: Revert #836 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/880
> * Update libOBS to 28.0.0-rc1 by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/878
</details>

<details><summary>0.12.0a45</summary>

> * Improve CMake build script(s) by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/813
> * ci: Add GCC back to the build by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/820
> * Add ClangCL, and update Windows SDK requirements by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/821
> * ci: Allow failures for experimental CI tests by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/824
> * ci: Use specific LLVM/Clang versions by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/825
> * project: Add support for multi stage release cycles by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/823
</details>

<details><summary>0.12.0a1</summary>

> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/778
> * ffmpeg-encoder/dnxhr: Added Avid DNxHR Encoder based on FFmpeg by @IceStormNG in https://github.com/Xaymar/obs-StreamFX/pull/784
> * New Crowdin updates by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/786
> * Refactor C++ wrappers for libOBS by @Xaymar in https://github.com/Xaymar/obs-StreamFX/pull/806
</details>

# System Requirements
<table style="table-layout: fixed">
  <tr>
    <th width="5%"><img width=10000000px height=1 /></th>
    <th width="5%"></th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-windows.svg" alt="Windows"><br>Windows</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-linux.svg" alt="Linux"><br>Linux</th>
    <th center width="30%"><img src="https://streamfx.xaymar.com/assets/platforms/platform-macos.svg" alt="MacOS"><br>MacOS</th>
  </tr>
  <tr>
    <th rowspan=2>OS</th>
    <th>Minimum</th>
    <td>Windows 10 21H2 (or newer)</td>
    <td>Ubuntu 20.04 (or equivalent)<br>Ubuntu 22.04 (or equivalent)</td>
    <td>MacOS 10.15 (or newer)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>Windows 10 21H2 (or newer)</td>
    <td>Ubuntu 20.04 (or equivalent)<br>Ubuntu 22.04 (or equivalent)</td>
    <td>MacOS 12 (or newer)</td>
  </tr>
  <tr>
    <th rowspan=4>CPU</th>
    <th rowspan=2>Minimum</th>
    <td colspan=3>
      <i>Any x86-64-v3 compatible:</i> <a href="https://en.wikipedia.org/wiki/Excavator_(microarchitecture)">AMD Excavator Series</a>, <a href="https://en.wikipedia.org/wiki/Haswell_(microarchitecture)">Intel Haswell Series</a><br/>
      <i>Required Instruction Sets:</i> x86-64-v2, AVX, BMI1, BMI2, F16C, FMA, MOVBE
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Recommended</th>
    <td colspan=3>
      <i>Any x86-64-v3 compatible:</i> <a href="https://en.wikipedia.org/wiki/Zen_2">AMD Zen 2 Series</a>, <a href="https://en.wikipedia.org/wiki/Comet_Lake_(microprocessor)">Intel Comet Lake Series</a><br/>
      <i>Required Instruction Sets:</i> x86-64-v2, AVX, BMI1, BMI2, F16C, FMA, MOVBE
    </td>
  </tr>
  <tr>
    <td colspan=2 style="display: none;"></td>
    <td>
      <i>Any ARM64-v8.5-A compatible:</i> <a href="https://en.wikipedia.org/wiki/Apple_M1">Apple M1 Series</a>
    </td>
  </tr>
  <tr>
    <th rowspan=2>Memory (RAM)</th>
    <th>Minimum</th>
    <td colspan=3>4 GiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td>8 GiB (free)</td>
    <td colspan=2>6 GiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>Video (GPU)</th>
    <th>Minimum</th>
    <td colspan=3>
      DirectX 11 / OpenGL 3 capable, Shader Model 4, 4 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>
      DirectX 12 / OpenGL 4 capable, Shader Model 5, 6 GiB VRAM (total)
    </td>
  </tr>
  <tr>
    <th rowspan=2>Disk Space</th>
    <th>Minimum</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>~50 MiB (free)</td>
  </tr>
  <tr>
    <th rowspan=2>OBS Version</th>
    <th>Minimum</th>
    <td colspan=3>28.0.0</td>
  </tr>
  <tr>
    <th>Recommended</th>
    <td colspan=3>28.0.0 or newer</td>
  </tr>
  <tr>
    <th colspan=2 center>Other</th>
    <td>
      <a href="https://aka.ms/vs/17/release/vc_redist.x64.exe">Microsoft Visual C/C++ Redistributable 2017-2022</a>
    </td>
    <td>
      Qt for obs-studio<br>
      FFmpeg for obs-studio<br>
      System Libraries must match chosen Ubuntu version!<br>
    </td>
    <td>
    </td>
  </tr>
</table>

<hr /><em>This discussion was created from the release <a href='https://github.com/Xaymar/obs-StreamFX/releases/tag/0.12.0a151'>StreamFX 0.12.0 Alpha 151</a>.</em>

**Comment by ypurpl** [comment]
I just installed it for use with the Steam OBS version, and it is not possible to install without admin rights. Other than that, it seems relatively stable (There was no prior installation).

**Reply by Xaymar** [reply]
System-wide installations always require Administrator rights.

**Reply by ypurpl** [reply]
Because of the `Program Files (x86)` folder, right?

**Reply by Xaymar** [reply]
No, StreamFX doesn't install there anymore. As the name already implies, System-wide modifies the "System", so it needs the rights to do so.

**Comment by PapillonDeFer** [comment]
Hi, which plugin should I download for windows? The .exe one says it won't run any advice greatly appreciated

**Reply by Xaymar** [reply]
You want the .exe. For installation guide see: https://github.com/Xaymar/obs-StreamFX/wiki/Installation

**Comment by PapillonDeFer** [comment]
Hey, 

Thank you so much for this, extremely helpful. I can’t get this to work, I’m on OBS 27.2.4 version. Is there a different link for that maybe? I couldn’t find it on your forum. 

Kind regards

Jo Selby
‬
E: ***@***.***
M: 07432 749 462

Sent from my iPhone using iCloud

On 6 Nov 2022, at 16:29, Xaymar ***@***.***> wrote:

﻿
You want the .exe. For installation guide see: https://github.com/Xaymar/obs-StreamFX/wiki/Installation

**Reply by Xaymar** [reply]
The stable version (v0.11.1) works with OBS 27.2.0 to 27.2.4. It should by default select the correct place.
